contracts/src/v0.8/ccip/tokenAdminRegistry/TokenPoolFactory/FactoryBurnMintERC20.sol

```diff
@@ -0,0 +1,289 @@
+// SPDX-License-Identifier: MIT
+pragma solidity 0.8.24;
+
+import {IGetCCIPAdmin} from "../../../ccip/interfaces/IGetCCIPAdmin.sol";
+import {IOwnable} from "../../../shared/interfaces/IOwnable.sol";
+import {IBurnMintERC20} from "../../../shared/token/ERC20/IBurnMintERC20.sol";
+
+import {OwnerIsCreator} from "../../../shared/access/OwnerIsCreator.sol";
+
+import {ERC20} from "../../../vendor/openzeppelin-solidity/v4.8.3/contracts/token/ERC20/ERC20.sol";
+import {IERC20} from "../../../vendor/openzeppelin-solidity/v4.8.3/contracts/token/ERC20/IERC20.sol";
+import {ERC20Burnable} from
+  "../../../vendor/openzeppelin-solidity/v4.8.3/contracts/token/ERC20/extensions/ERC20Burnable.sol";
+import {IERC165} from "../../../vendor/openzeppelin-solidity/v4.8.3/contracts/utils/introspection/IERC165.sol";
+import {EnumerableSet} from "../../../vendor/openzeppelin-solidity/v4.8.3/contracts/utils/structs/EnumerableSet.sol";
+
+/// @notice A basic ERC20 compatible token contract with burn and minting roles.
+/// @dev The constructor has been modified to support the deployment pattern used by a factory contract.
+/// @dev The total supply can be limited during deployment.
+contract FactoryBurnMintERC20 is IBurnMintERC20, IGetCCIPAdmin, IERC165, ERC20Burnable, OwnerIsCreator {
+  using EnumerableSet for EnumerableSet.AddressSet;
+
+  error SenderNotMinter(address sender);
+  error SenderNotBurner(address sender);
+  error MaxSupplyExceeded(uint256 supplyAfterMint);
+
+  event MintAccessGranted(address indexed minter);
+  event BurnAccessGranted(address indexed burner);
+  event MintAccessRevoked(address indexed minter);
+  event BurnAccessRevoked(address indexed burner);
```

Comment:
q: why use `indexed` if the even has a single field?

Reply:
Good question. Not sure but that was how our other token files implemented that event so I didn't see a reason to modify it.